soh/src/code/z_parameter.c

```diff
@@ -1493,7 +1493,20 @@ void Inventory_SwapAgeEquipment(void) {
             gSaveContext.equips.equipment = gSaveContext.childEquips.equipment;
             gSaveContext.equips.equipment &= 0xFFF0;
             gSaveContext.equips.equipment |= 0x0001;
+        } else { //This situation should only really apply in rando where adult has never been child before
+            if (gSaveContext.n64ddFlag && (1 << 0 & gSaveContext.inventory.equipment) == 0) {
```

Comment:
This might just be preference, but I think it'd be better to have the else statement check for the n64ddFlag so we're absolutely sure it won't touch vanilla. Would also clarify the intention a little bit.